File: VB/ViewModels/AppViewModel.vb
Imports System.Threading.Tasks
Imports DevExpress.Mvvm
Imports DevExpress.Mvvm.POCO
Imports TodoApp.Data

Namespace TodoApp.ViewModels
    Public Class AppViewModel
        Shared Sub New()
            ' Registering global service for data-storage
            ServiceContainer.Default.RegisterService(New InMemoryRepository())
        End Sub
        ' This is ViewModel for our Application
        Public Sub New()
            Title = "Getting started witn MVVM - Todo App (Visual Basic)"
        End Sub
        ' This is our Application's Title
        Public ReadOnly Property Title As String
        Protected ReadOnly Property NavigationService As INavigationService
            Get
                Return GetService(Of INavigationService)()
            End Get
        End Property
        Public Async Function OnShown() As Task
            ' Show Logo screen
            NavigationService.Navigate(NameOf(Views.AppLogo), Nothing, Me, False)
            ' Wait some time before showing ItemsView
            Await Task.Delay(2000)
            ' Show Items screen
            NavigationService.Navigate(NameOf(Views.ItemsView), Nothing, Me, True)
        End Function
    End Class
End Namespace

File: CS/ViewModels/AppViewModel.cs
namespace TodoApp.ViewModels {
    using System.Threading.Tasks;
    using DevExpress.Mvvm;
    using DevExpress.Mvvm.POCO;
    using TodoApp.Data;

    public class AppViewModel {
        static AppViewModel() {
            // Registering global service for data-storage
            ServiceContainer.Default.RegisterService(new InMemoryRepository());
        }
        // This is ViewModel for our Application
        public AppViewModel() {
            Title = "Getting started witn MVVM - Todo App (C#)";
        }
        // This is our Application's Title
        public string Title {
            get;
            private set;
        }
        protected INavigationService NavigationService {
            get { return this.GetService<INavigationService>(); }
        }
        public async Task OnShown() {
            // Show Logo screen
            NavigationService.Navigate(nameof(Views.AppLogo), null, this, false);
            // Wait some time before showing ItemsView
            await Task.Delay(2000);
            // Show Items screen
            NavigationService.Navigate(nameof(Views.ItemsView), null, this, true);
        }
    }
}